.github/dependabot.yml

```diff
@@ -32,3 +32,7 @@ updates:
     ignore:
       - dependency-name: "*"
         update-types: [ "version-update:semver-major" ]
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "monthly"
```

Comment:
So... while switching to SHAs is a good thing for stability (we'll always run the same code), having dependabot upgrade the actions regularly means we'll regularly pull in new versions, so we'll regularly be running new code, in particular in our releases.

In short we're trading the (current) risk of someone re-defining a tag in a GitHub action repository and inserting malicious code in our build, for the (new) risk of someone pushing and tagging a new tag in a GitHub action repository, to have Dependabot switch our build to this, and eventually inserting malicious code in our build.

I guess it's... slightly better :grimacing:

Reply:
yeah ... I haven't thought about it from that perspective... 
my thinking here was that we wouldn't want to keep the versions stale for a long time so we don't end up using a deprecated version at some point 🙈 
let me know how you'd prefer to handle it 😃

Reply:
Frankly I don't know.

Switching to SHA, and only doing that, seems fine.

Upgrading GitHub actions is fine as well, until you get pwned. But then staying on an older version of any software comes with security risks, so it's not like we're safe in any case.

The ideal solution would be to update only to trusted versions of artifacts. Either because we trust the whole namespace (`actions/*`, `redhat-actions/*`), or because we manually checked the source code of the new version (yeah right, who am I kidding).

Maybe enable dependabot for select namespaces that are reputable enough that we're willing to take the risk of trusting their new releases?

Reply:
> actions/*, redhat-actions/*

added these two, looking at the remaining ones: we have `azure` and `dependabot`

Reply:
Let's start with that?

Reply:
yeah and see how it'll go 👍🏻 😃